item-counter-app/src/App.js

```diff
@@ -1,25 +1,80 @@
-import logo from './logo.svg';
-import './App.css';
-
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+import React, { Component } from "react";
+import "./App.css";
+import Counters from "./component/Counters";
+import Navbar from "./component/navbar";
+
+class App extends Component {
+  state = {
+    counters: [
+      { id: 1, value: 0 },
+      { id: 2, value: 0 },
+      { id: 3, value: 0 },
+    ],
+  };
+
+  handleDelete = (id) => {
+    this.setState({
+      counters: this.state.counters.filter((counter) => counter.id != id),
+    });
+  };
+
+  handleAdd = () => {
+    const { counters } = this.state;
+    counters.push({
+      id: counters.length ? counters[counters.length - 1].id + 1 : 1,
+      value: 0,
+    });
```

Comment:
You are directly modifying a variable in state. This is a bad practice in React.

You have to make a copy of `counters` **first** and then modify that.

Reply:
resolved